### PR TITLE
[3.9] bpo-45866: pegen strips directory of "generated from" header (GH-29777) (GH-29792)

### PR DESCRIPTION
**Pull request by vstinner**
"make regen-all" now produces the same output when run from a
directory other than the source tree: when building Python out of the
source tree.

(cherry picked from commit 253b7a0a9fef1d72a4cb87b837885576e68e917c)
(cherry picked from commit b6defde2afe656db830d6fedf74ca5f6225f5928)

https://bugs.python.org/issue45866

### CODE REVIEW

File: Misc/NEWS.d/next/Build/2021-11-25-13-53-36.bpo-45866.ZH1W8N.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+``make regen-all`` now produces the same output when run from a directory
+other than the source tree: when building Python out of the source tree.
+pegen now strips directory of the "generated by pygen from <FILENAME" header
```
Comment (encukou) [inline Misc/NEWS.d/next/Build/2021-11-25-13-53-36.bpo-45866.ZH1W8N.rst:3]:
```suggestion
pegen now strips directory of the "generated by pygen from <FILENAME>" header
```

Comment (hroncok) [inline Misc/NEWS.d/next/Build/2021-11-25-13-53-36.bpo-45866.ZH1W8N.rst:3]: Unclosed `<`

Comment (vstinner) [reply Misc/NEWS.d/next/Build/2021-11-25-13-53-36.bpo-45866.ZH1W8N.rst:3]: Oh, that's a backport of a change which already landed in main and 3.10 branches :-( I prefer to keep the typo. The typo can be fixed in a separated PR, but I'm not really motivated to write a PR just for that :-p

Comment (vstinner) [reply Misc/NEWS.d/next/Build/2021-11-25-13-53-36.bpo-45866.ZH1W8N.rst:3]: Well, I created PR to fix this important typo :-) https://github.com/python/cpython/pull/29798/files

cc @hroncok

**Pull request merged**